### PR TITLE
more readable Buffer.compare

### PR DESCRIPTION
**Pull request by fanatid**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -285,17 +285,8 @@ Buffer.compare = function compare (a, b) {
...
-    if (a[i] !== b[i]) break
-
-    ++i
-  }
-
-  if (i !== len) {
-    x = a[i]
-    y = b[i]
+  for (var i = 0, len = Math.min(x, y); i < len; ++i) {
+    if (a[i] !== b[i]) return a[i] > b[i] ? 1 : -1
```
Comment (feross) [inline index.js]: Would prefer this to be flipped to use `<` like the rest of the comparators in the function. Will do that post-merge.

**Comment by feross** [comment]
LGTM

File: index.js (hunk 1 above)
Comment (feross) [inline index.js]: Also, I think that early returns were intentionally avoided in this function for perf reasons. Going to remove it. But your other changes are fine.

**Comment by fanatid** [comment]
Thank you.
Do you have any benchmarks about performance for early return?

**Comment by feross** [comment]
Not within the buffer project, but if you google it you can learn more.

**Comment by fanatid** [comment]
It's specific benchmark, google will gives nothing.
```shell
$ uname -a
Linux localhost.localdomain 4.4.5-300.fc23.x86_64 #1 SMP Thu Mar 10 17:54:44 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
$ node -v
v5.7.1
$ cat bench2.js 
var benchmark = require('benchmark')
var createHash = require('create-hash')

var buf1 = new Buffer(1024)
var buf2 = new Buffer(buf1)
buf2[buf2.length - 10] += 1

function cmp1 () {
  var x = buf1.length
  var y = buf2.length

  for (var i = 0, len = Math.min(buf1.length, buf2.length); i < len; ++i) {
    if (buf1[i] !== buf2[i]) return buf1[i] > buf2[i] ? 1 : -1
  }

  if (x < y) return -1
  if (y < x) return 1
  return 0
}


function cmp2 () {
  var x = buf1.length
  var y = buf2.length

  for (var i = 0, len = Math.min(buf1.length, buf2.length); i < len; ++i) {
    if (buf1[i] !== buf2[i]) {
      x = buf1[i]
      y = buf2[i]
      break
    }
  }

  if (x < y) return -1
  if (y < x) return 1
  return 0
}

new benchmark.Suite()
  .on('cycle', (event) => console.log(String(event.target)))
  .add('break', () => cmp2)
  .add('early exit', () => cmp1)
  .run()
$ node bench2.js 
early exit x 51,004,415 ops/sec ±1.53% (81 runs sampled)
break x 51,235,002 ops/sec ±1.04% (88 runs sampled)
$ node bench2.js 
early exit x 51,907,082 ops/sec ±1.52% (84 runs sampled)
break x 52,595,909 ops/sec ±1.28% (85 runs sampled)
$ node bench2.js 
early exit x 51,724,788 ops/sec ±1.47% (83 runs sampled)
break x 50,746,237 ops/sec ±1.68% (86 runs sampled)
$ node bench2.js 
break x 52,120,941 ops/sec ±1.59% (83 runs sampled)
early exit x 51,695,328 ops/sec ±1.61% (87 runs sampled)
$ node bench2.js 
break x 51,728,658 ops/sec ±1.49% (83 runs sampled)
early exit x 52,043,099 ops/sec ±1.44% (86 runs sampled)
$ node bench2.js 
break x 50,741,374 ops/sec ±1.59% (82 runs sampled)
early exit x 51,561,304 ops/sec ±1.02% (87 runs sampled)
```

looks like jitter

**Pull request merged**